### PR TITLE
Export RedisClientType and RedisClusterType types

### PR DESCRIPTION
**Pull request by kjlis**
### Description

Export RedisClientType and RedisClusterType types

Fixes redis/node-redis#1673

---

### Checklist

- [x] Does `npm test` pass with this change (including linting)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)? - N/A

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,10 +1,14 @@
 import RedisClient from './lib/client';
 import RedisCluster from './lib/cluster';
 
+export { RedisClientType } from './lib/client';
```
Comment (Jesus-Escalona) [inline index.ts:4]: Awesome thanks!

**Comment by davelosert** [comment]
@kjlis : We kind of need this feature as well - but it seems like in this PR there are some merge conflicts currently.
Do you have the time to fix them?

If not I can offer to take it over from here, if you don't mind of course 🙂

**Comment by kjlis** [comment]
@davelosert I'll rebase with the latest changes and update the PR

**Comment by leibale** [comment]
@kjlis can you please remove `packages/test-utils/package-lock.json`? other than that LGTM.

File: index.ts [hunk 2]
```diff
@@ -1,7 +1,6 @@
 import { createClient as _createClient, createCluster as _createCluster } from '@node-redis/client';
 import { RedisScripts } from '@node-redis/client/dist/lib/commands';
-import { RedisClientOptions, RedisClientType } from '@node-redis/client/dist/lib/client';
-import { RedisClusterOptions, RedisClusterType } from '@node-redis/client/dist/lib/cluster';
+import { RedisClientOptions, RedisClientType, RedisClusterOptions, RedisClusterType } from '@node-redis/client';
```
Comment (leibale) [inline index.ts:3]: please merge this import with the first import

File: packages/test-utils/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,3927 @@
+{
```
Comment (leibale) [inline packages/test-utils/package-lock.json:1]: please remove this file

**Review by leibale — CHANGES_REQUESTED** [review]

File: index.ts (hunk 2 above)
Comment (kjlis) [reply index.ts]: Merged.

File: packages/test-utils/package-lock.json (hunk 3 above)
Comment (kjlis) [reply packages/test-utils/package-lock.json]: Removed.

**Comment by kjlis** [comment]
@leibale the workflow failed with the timeout error:
```
 "before all" hook for "scripts":
1564
     Error: Timeout of 30000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```
Is it a transient error? If so, can we rerun the workflow?

**Review by leibale — APPROVED** [review]

**Comment by leibale** [comment]
@kjlis thanks! :)

**Pull request merged**